lib/tools.js

```diff
@@ -2223,14 +2223,14 @@ function maybeCallback(callback, ...args) {
 /**
  * Checks if the given callback is a function and if so calls it with the given error and parameter immediately, else a resolved or rejected Promise is returned
  *
- * @param {function(Error=, ...any=):void|undefined} callback - callback function to be executed
- * @param {any} [error] - error which will be used by the callback function. If callback is not a function and
+ * @param {((error: Error, ...args: any[]) => void) | null | undefined} callback - callback function to be executed
+ * @param {Error | string | null | undefined} error - error which will be used by the callback function. If callback is not a function and
```

Comment:
It makes no sense to call this method without an argument for Error IMO. Explicitly having to pass undefined or null as the error makes this less error-prone.